lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,45 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+
+            # Use the return value of the block unless it is nil.
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
+            else
+              self._value = value
+            end
           end
 
           def href(value)
             self._href = value
+            nil
           end
 
           def meta(value)
             self._meta = value
+            nil
+          end
+
+          def as_json
+            _value || to_hash
           end
 
+          protected
+
+          attr_accessor :_href, :_meta, :_value
+          attr_reader :object, :scope
+
+          private
+
           def to_hash
```

Comment:
`block_value || hash_value` ? `to_hash` doesn't really jive well with how I think of the method